R/kernel.r

```diff
@@ -104,10 +104,14 @@ new_reply = function(msg_type, parent_msg) {
         metadata      = namedlist())
 },
 
-send_response = function(msg_type, parent_msg, socket_name, content) {
+send_response = function(msg_type, parent_msg, socket_name, content, metadata = NULL) {
```

Comment:
When would you set metadata to something else than `request_metadata.get()`?

IMO there are tow "shorter" ways to structure this PR:

* define the function as `send_response = function(msg_type, parent_msg, socket_name, content, metadata = request_metadata.get())` and remove all the above `request_metadata.get()` calls
* If it is just a echoing back of the metadata don't redefine the function and just call `request_metadata.get()` in the `send_response` body.